### PR TITLE
feat(mz) Improve regions in CLI

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

- [x] Add `regions status` command
- [x] Format `regions list` command
- [x] Add loading to the `regions enable` command (Using `indicatif` crate)
- [x] Disable `regions delete` command
- [x] Refactor: `.clones` -> reference borrowing  
- [x] Refactor: Panics -> error message + `process.exit()`
- [x] Improved + commented code  

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

File: src/mz/src/regions.rs [hunk 1]
```diff
@@ -131,39 +110,49 @@ pub(crate) async fn cloud_provider_region_details(
...
     }
 
-    regions
+    cloud_providers_and_regions
 }
 
+/**
+ * List all the available cloud providers.
+ * E.g.: [us-east-1, eu-west-1]
+ */
```
Comment (benesch) [inline src/mz/src/regions.rs:152]: Rustdoc uses `///`, not `/*** ... */`, here and throughout!

File: src/mz/src/regions.rs [hunk 2]
```diff
@@ -177,3 +166,85 @@ pub(crate) async fn list_cloud_providers(
...
+    println!("SQL address: \t{}", region.environmentd_pgwire_address);
+    // Remove port from url
+    println!(
+        "HTTPS address: \thttps://{}",
+        &region.environmentd_https_address[0..region.environmentd_https_address.len() - 4]
+    );
+}
+
+// ------------------------------------------------------------------------
+// Delete is currently disabled. Preserving the code for once is available.
```
Comment (benesch) [inline src/mz/src/regions.rs:211]:
```suggestion
// Delete is currently disabled. Preserving the code for once is available.
```

Delete is actually always going to be disabled (it'll be a support-only operation), so I think we should just pull the plug on this now.

**Review by benesch — APPROVED** [review]
Sorry for the slowness! Didn't look too close, but LGTM!

**Comment by joacoc** [comment]
Applying the documentation changes and removing the delete in [the other PR](https://github.com/MaterializeInc/materialize/pull/14118) to make the merge simpler

**Pull request merged**